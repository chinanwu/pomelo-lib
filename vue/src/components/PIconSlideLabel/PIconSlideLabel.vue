<template>
  <div class="PIconSlideLabel" :data-label="label">
    <a
      class="PIconSlideLabel__link"
      v-if="link"
      :href="link"
      :title="linkTitle"
      :aria-label="label"
    >
      <img
        class="PIconSlideLabel__icon"
        :class="iconClass"
        :src="iconUrl"
        :alt="alt"
      />
    </a>
    <img
      v-else
      class="PIconSlideLabel__icon"
      :class="iconClass"
      :src="iconUrl"
      :alt="alt"
      :aria-label="label"
    />
  </div>
</template>

<script>
export default {
  name: "PIconSlideLabel",
  props: {
    iconUrl: {
      // Must be passed in with require()
      type: String,
      required: true
    },
    link: {
      type: String,
      default: null
    },
    linkTitle: {
      type: String,
      default: null
    },
    iconClass: {
      type: String,
      default: null
    },
    label: {
      type: String,
      required: true
    },
    alt: {
      type: String,
      required: true
    }
  }
};
</script>

<style lang="less" scoped>
.PIconSlideLabel {
  display: flex;

  &::after {
    content: attr(data-label);
    display: flex;
    align-items: center;
    transform: translateX(-1rem);
    opacity: 0;
    transition: all 0.2s;
    margin-left: 1rem;
  }

  &:hover::after {
    transform: translateX(0);
    opacity: 1;
  }
}

.PIconSlideLabel__link {
  display: contents;
}
</style>

<!-- TODO -->
<!-- - Prefer No Motion users - set so no transform -->
