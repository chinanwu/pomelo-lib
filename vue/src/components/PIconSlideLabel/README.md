# PIconSlideLabel

A component that displays an icon with a label that slides out upon hovering over the icon.

[Click here to see a demo of these buttons in action](https://chinanwu.github.io/pomelo-lib/vue/index.html#piconslidelabel)

## Example
```html
<PIconSlideLabel 
    :iconUrl="require('path/relative/to/file/using/this/component')" 
    label="Label" 
    alt="Image description"
>
</PIconSlideLabel>
```
Alternatively, you can use the hyphenated component name.
```html
<p-icon-slide-label></p-icon-slide-label>
```

## Props

| Prop | Type |  Default | Required | Description | Validation |
| ---- | ---- | -------- | -------- | ----------- | ---------- |
| iconUrl | String | N/A | true | The path to the icon img, relative to the component using it. Must be wrapped with `require()` to work. | None |
| alt | String | N/A | true | Description of the icon. | None |
| label | String | N/A | true | The label that slides out upon hovering over the icon | None |
| link | String | null | false | URL to go to when click on icon | None |
| linkTitle | String | null | false | A title for the link | None |
| iconClass | String | null | false | Style to apply to icon | None |

## Best practices

- TBA 

## See an issue?
- Please create an issue [here](https://github.com/chinanwu/pomelo-lib/issues). Indicate the component and the issue, thank you! 
- Alternatively, if you fix the issues, you can create a [pull request](https://github.com/chinanwu/pomelo-lib/pulls)! 
