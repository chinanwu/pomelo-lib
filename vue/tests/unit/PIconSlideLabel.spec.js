import { shallowMount } from "@vue/test-utils";

import PIconSlideLabel from "@/components/PIconSlideLabel/PIconSlideLabel";

describe("PIconSlideLabel", () => {
  describe("rendering", () => {
    it("renders PIconSlideLabel", () => {
      const wrapper = shallowMount(PIconSlideLabel, {
        props: {
          iconUrl: "fake/picture.png",
          alt: "fake",
          label: "Label"
        }
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it("renders linked PIconSlideLabel", () => {
      const wrapper = shallowMount(PIconSlideLabel, {
        props: {
          iconUrl: "fake/picture.png",
          alt: "fake",
          label: "Label",
          link: "fake.com",
          linkTitle: "Fake site"
        }
      });
      expect(wrapper.element).toMatchSnapshot();
    });

    it("renders PIconSlideLabel with sizes", () => {
      const wrapper = shallowMount(PIconSlideLabel, {
        props: {
          iconUrl: "fake/picture.png",
          alt: "fake",
          label: "Label",
          iconHeight: 12,
          iconWidth: 12
        }
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
