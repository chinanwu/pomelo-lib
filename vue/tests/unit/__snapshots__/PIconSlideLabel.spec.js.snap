// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PIconSlideLabel rendering renders PIconSlideLabel 1`] = `
<div
  class="PIconSlideLabel"
  data-label="Label"
>
  <img
    alt="fake"
    aria-label="Label"
    class="PIconSlideLabel__icon"
    data-height="32"
    data-width="32"
    src="fake/picture.png"
    style="width: 32px; height: 32px;"
  />
</div>
`;

exports[`PIconSlideLabel rendering renders PIconSlideLabel with sizes 1`] = `
<div
  class="PIconSlideLabel"
  data-label="Label"
>
  <img
    alt="fake"
    aria-label="Label"
    class="PIconSlideLabel__icon"
    data-height="12"
    data-width="12"
    src="fake/picture.png"
    style="width: 12px; height: 12px;"
  />
</div>
`;

exports[`PIconSlideLabel rendering renders linked PIconSlideLabel 1`] = `
<div
  class="PIconSlideLabel"
  data-label="Label"
>
  <a
    aria-label="Label"
    class="PIconSlideLabel__link"
    href="fake.com"
    title="Fake site"
  >
    <img
      alt="fake"
      class="PIconSlideLabel__icon"
      src="fake/picture.png"
      style="width: 32px; height: 32px;"
    />
  </a>
</div>
`;
